core/src/mindustry/world/blocks/experimental/BlockForge.java

```diff
@@ -41,7 +41,7 @@ public BlockForge(String name){
     public void setBars(){
         super.setBars();
 
-        bars.add("progress", entity -> new Bar("bar.progress", Pal.ammo, () -> ((BlockForgeBuild)entity).progress));
+        bars.add("progress", (BlockForgeBuild entity) -> new Bar("bar.progress", Pal.ammo, () -> entity.payload != null ? 1f : entity.recipe == null ? 0f : (float)(entity.progress / entity.recipe.buildCost)));
```

Comment:
accept this suggestion in case you want the progress bar to reset to 0 before the block finishes moving out:

```suggestion
        bars.add("progress", (BlockForgeBuild entity) -> new Bar("bar.progress", Pal.ammo, () -> entity.recipe == null ? 0f : (float)(entity.progress / entity.recipe.buildCost)));
```